pkg/pmem-csi-operator/controller/deployment/deployment_controller.go

```diff
@@ -85,8 +213,10 @@ type ReconcileDeployment struct {
 	// container image used for deploying the operator
 	containerImage string
 	// known deployments
-	deployments    map[string]*pmemcsiv1alpha1.Deployment
-	reconcileHooks map[ReconcileHook]struct{}
+	deployments      map[string]*pmemcsiv1alpha1.Deployment
+	deploymentsMutex sync.Mutex
+	reconcileMutex   sync.Mutex
```

Comment:
Please add comments explaining what is protected by these mutexes and why that is necessary. `deploymentsMutex` seems relatively straightforward (protect concurrent access to `deployments`) but the logic around locking `reconcileMutex` isn't.

Reply:
Added the field comments.